### ISSUE TITLE
什么时候开源c++版本？

### ISSUE THREAD

**Issue by avcodecs** [cloudwego/shmipc-go#10]

**Comment by goodbye-babyer** [comment]
也许在2023 Q3

**Comment by qingtiandalaoye** [comment]
标记一下，等c++

**Comment by avcodecs** [comment]
look forward to it. I want to use it in embedded devices.

**Comment by goodbye-babyer** [comment]
> look forward to it. I want to use it in embedded devices.

arm arch?

**Comment by avcodecs** [comment]
yes



---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 日期 | 2023年04月24日 12:01 |
| 收件人 | ***@***.***> |
| 抄送至 | ***@***.***>***@***.***> |
| 主题 | Re: [cloudwego/shmipc-go] 什么时候开源c++版本？ (Issue #10) |

look forward to it. I want to use it in embedded devices.

arm arch?

**Comment by GuangmingLuo** [comment]
如果还有其它用户期待使用 C++ 版本，欢迎在这个 issue 下面留言，可以加速项目开源进程。

**Comment by kolinfluence** [comment]
@GuangmingLuo yeah me. pls make it compatible with shmipc-go so both can talk to each other!

**Comment by GuangmingLuo** [comment]
> @GuangmingLuo yeah me. pls make it compatible with shmipc-go so both can talk to each other!

Copy that

**Comment by kolinfluence** [comment]
@GuangmingLuo possible to do a c version as well if it's easy to convert to from c++ i guess. not trying to be funny but cgo, c seems to make sense here too. just saying.

**Comment by kolinfluence** [comment]
@GuangmingLuo it's working very well with golang (just uses a lot more memory now). it's really good. maybe also do for rust and all will be great. it's ok to tweak or enhance new version to be compatible with all major performance programming language.

thanks for it. cant thank you enough

**Comment by haoyixin** [comment]
> 如果还有其它用户期待使用 C++ 版本，欢迎在这个 issue 下面留言，可以加速项目开源进程。

other than being used for comm with cpp sidecar, I can't think of any other use cases 😂

**Comment by hiqsociety** [comment]
is there an ETA on this? i could be waiting 10 years... just curious

**Comment by GuangmingLuo** [comment]
> @GuangmingLuo it's working very well with golang (just uses a lot more memory now). it's really good. maybe also do for rust and all will be great. it's ok to tweak or enhance new version to be compatible with all major performance programming language.
> 
> thanks for it. cant thank you enough

Thanks for your acknowledgement, pls leave a message here: https://github.com/cloudwego/shmipc-go/issues/25  @kolinfluence

**Comment by GuangmingLuo** [comment]
> look forward to it. I want to use it in embedded devices.

@avcodecs hi, will you describe it in detail about how will you use it in embedded devices. Which company will use it ?

**Comment by Czhazha** [comment]
1

**Comment by goodbye-babyer** [comment]
> 1

what is your scenario

**Comment by geata** [comment]
同样期待c++版本

**Comment by goodbye-babyer** [comment]
> 同样期待c++版本

优先级目前不太高 方便说下你们的应用场景吗

**Comment by avcodecs** [comment]
嵌入式设备的音频io与传输之间的共享缓存。进程间通信。



---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 日期 | 2024年01月31日 16:02 |
| 收件人 | ***@***.***> |
| 抄送至 | ***@***.***>***@***.***> |
| 主题 | Re: [cloudwego/shmipc-go] 什么时候开源c++版本？ (Issue #10) |

同样期待c++版本

优先级目前不太高 方便说下你们的应用场景吗

**Comment by GuangmingLuo** [comment]
大家评论的时候，可以留下公司信息 + 应用场景，这样我们才能拿到明确的需求去排期开源。

**Comment by sprappcom** [comment]
looking forward to rust version also, though been waiting for c++ version for so long. hope it can be interlanguage communication possible.

**Comment by GuangmingLuo** [comment]
When you ask for open source requests, please leave your **company information + application scenarios**, so that we can get a clear demand to schedule it. Thanks.

**Comment by maple0575** [comment]
期待C++版本

**Comment by kolinfluence** [comment]
@GuangmingLuo c will do fine.

**Comment by GuangmingLuo** [comment]
Waiting for: **company information + application scenarios**

**Comment by chongtianfeiyu** [comment]
期待c++版本

**Comment by Yanjiachan** [comment]
期待

**Comment by bigolcy** [comment]
我们实现了C++的版本，在实际场景（上游C++服务，下游envoy）测试，效果不及预期（1M包以下的场景几乎没有提升），请问可能是哪个方面出了问题？

**Comment by lixai** [comment]
期待C++版本开源，主要用于车载嵌入式场景中。可以尽快开源，这样大家也可以提供一些patch改进项目。

**Comment by ziyangfu** [comment]
> 我们实现了C++的版本，在实际场景（上游C++服务，下游envoy）测试，效果不及预期（1M包以下的场景几乎没有提升），请问可能是哪个方面出了问题？

你好，请问开源嘛？

**Comment by ziyangfu** [comment]
> 期待C++版本开源，主要用于车载嵌入式场景中。可以尽快开源，这样大家也可以提供一些patch改进项目。

嗨咯，同车载领域，有没有兴趣一起写？

**Comment by dancsmshenry** [comment]
> > 期待C++版本开源，主要用于车载嵌入式场景中。可以尽快开源，这样大家也可以提供一些patch改进项目。
> 
> 嗨咯，同车载领域，有没有兴趣一起写？

感觉逻辑上就是 unix socket + shared memory，但是比较好奇的是该怎么把这个逻辑嵌入到 someip/dds 中，请问下同学对这方面有啥思路吗

**Comment by ziyangfu** [comment]
> > > 期待C++版本开源，主要用于车载嵌入式场景中。可以尽快开源，这样大家也可以提供一些patch改进项目。
> > 
> > 
> > 嗨咯，同车载领域，有没有兴趣一起写？
> 
> 感觉逻辑上就是 unix socket + shared memory，但是比较好奇的是该怎么把这个逻辑嵌入到 someip/dds 中，请问下同学对这方面有啥思路吗

我有一个业余项目 [TinyNetFlow](https://github.com/ziyangfu/TinyNetFlow)，计划实现Adaptive AUTOSAR的部分功能，还在进行中，可以看看，或者一起写？

**Comment by gufuguang** [comment]
C++版本开源了吗？用在工业领域

**Comment by PureWhiteWu** [comment]
可以考虑用 shmipc-rs

**Comment by dwh110** [comment]
期待C++版本的

**Comment by dwh110** [comment]
> 我们实现了C++的版本，在实际场景（上游C++服务，下游envoy）测试，效果不及预期（1M包以下的场景几乎没有提升），请问可能是哪个方面出了问题？

效果不及预取的原因找到了吗？

**Comment by ggq89** [comment]
我也很期待C++版本开源，加油
